src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -39,6 +57,36 @@ class HttpGoogleCloudStorageDAO(clientSecretsJson: String, dataStoreFactory: Dat
      .build()
   }
 
+  override def createGoogleGroup(userId: String, accessLevel: String, workspaceName: WorkspaceName, bucketName: String) = {
+    val credential = new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(clientSecrets.getDetails.get("client_email").toString)
+      .setServiceAccountScopes(directoryScopes)
+      .setServiceAccountUser(clientSecrets.getDetails.get("sub_email").toString)
+      .setServiceAccountPrivateKeyFromP12File(new java.io.File(gcsConfig.getString("pathToP12"))) //add this file to jenkins?
+      .build()
+    val appsDomain = gcsConfig.getString("appsDomain")
+    val groupsPrefix = gcsConfig.getString("groupsPrefix")
+
+    val directory = new Directory.Builder(httpTransport, jsonFactory, credential).setApplicationName("firecloud:rawls").build()
+    val group = new Group().setEmail(s"${groupsPrefix}-${workspaceName.namespace}_${workspaceName.name}-${accessLevel}@${appsDomain}").setName(s"FireCloud ${workspaceName.namespace}/${workspaceName.name} ${accessLevel}")
```

Comment:
Can we put this into separate functions for generating email and name for a given workspace + access level?